File: frontend/src/types/media.ts
export interface EnhanceRequest {
  text: string;
  language?: string;
  generate_image_prompts?: boolean;
}

export interface EnhanceResponse {
  enhanced_text: string;
  image_prompts: ScriptPromptItem[];
}

export interface GenerateScriptRequest {
  url: string;
  language?: string;
}

export interface ScriptPromptItem {
  timestamp: string;
  cue: string;
  prompt: string;
}

export interface GenerateScriptResponse {
  script: string;
  image_prompts: ScriptPromptItem[];
  source_chars: number;
}

export interface ShortPromptItem {
  cue: string;
  prompt: string;
}

export interface WhiskPromptItem {
  cue: string;
  prompt: string;
}

export interface TimelineSegment {
  slot_index: number;
  asset_kind: 'video' | 'image';
  duration_seconds: number;
  motion_preset: string;
  overlay_text: string | null;
}

export interface ProductionPack {
  template_id: string;
  flow_video_prompt: string;
  whisk_image_prompts: WhiskPromptItem[];
  timeline_segments: TimelineSegment[];
  caption_text: string;
  cta_text: string;
}

export interface ShortItem {
  id: string;
  title: string;
  hook: string;
  script: string;
  cta: string;
  image_prompts: ShortPromptItem[];
  broll_keywords: string[];
  production_pack: ProductionPack | null;
}

export interface GenerateShortRequest {
  script: string;
  count: number;
  duration_seconds: number;
  language?: string;
}

export interface GenerateShortResponse {
  shorts: ShortItem[];
  count: number;
  duration_seconds: number;
}

export interface AssetSearchRequest {
  keywords: string[];
  collection: string;
}

export interface BrollClip {
  title: string;
  identifier: string;
  license: string;
  download_url: string;
  thumb: string;
  preview_url: string;
}

export interface AssetSearchResponse {
  clips: BrollClip[];
}

export interface BrollDownloadRequest {
  url: string;
}

export interface BrollDownloadResponse {
  local_path: string;
  filename: string;
}

export interface AssembleJobResponse {
  job_id: string;
}

export interface AssembleStatusResponse {
  status: 'processing' | 'done' | 'error';
  progress: number;
  download_url?: string | null;
  error?: string | null;
}

export interface StreamProgress {
  current: number;
  total: number;
  label: string;
}

export interface TtsLogItem {
  id: string;
  level: 'info' | 'ok' | 'warn' | 'error';
  message: string;
}

File: app/schemas/media.py
from pydantic import BaseModel, Field, HttpUrl


class EnhanceRequest(BaseModel):
    api_key: str | None = None
    text: str = Field(min_length=1)
    language: str = "pt"
    generate_image_prompts: bool = False


class ScriptPromptItem(BaseModel):
    timestamp: str = "00:00"
    cue: str = ""
    prompt: str


class EnhanceResponse(BaseModel):
    enhanced_text: str
    image_prompts: list[ScriptPromptItem] = Field(default_factory=list)


class GenerateTTSRequest(BaseModel):
    api_key: str | None = None
    text: str = Field(min_length=1)
    voice: str = "Charon"
    language: str = "pt"


class GenerateTTSStreamRequest(GenerateTTSRequest):
    session_id: str | None = None


class GenerateScriptRequest(BaseModel):
    api_key: str | None = None
    url: HttpUrl
    language: str = "pt"


class GenerateScriptResponse(BaseModel):
    script: str
    image_prompts: list[ScriptPromptItem]
    source_chars: int


class ShortPromptItem(BaseModel):
    cue: str = ""
    prompt: str


class WhiskPromptItem(BaseModel):
    cue: str = ""
    prompt: str


class TimelineSegment(BaseModel):
    slot_index: int
    asset_kind: str
    duration_seconds: int
    motion_preset: str
    overlay_text: str | None = None


class ProductionPack(BaseModel):
    template_id: str
    flow_video_prompt: str
    whisk_image_prompts: list[WhiskPromptItem]
    timeline_segments: list[TimelineSegment]
    caption_text: str
    cta_text: str


class ShortItem(BaseModel):
    id: str
    title: str
    hook: str
    script: str
    cta: str
    image_prompts: list[ShortPromptItem]
    broll_keywords: list[str]
    production_pack: ProductionPack | None = None


class GenerateShortRequest(BaseModel):
    api_key: str | None = None
    script: str = Field(min_length=500)
    count: int = 3
    duration_seconds: int = 60
    language: str = "pt"


class GenerateShortResponse(BaseModel):
    shorts: list[ShortItem]
    count: int
    duration_seconds: int


class AssetSearchRequest(BaseModel):
    keywords: list[str] = Field(min_length=1)
    collection: str = "prelinger"


class BrollClip(BaseModel):
    title: str = ""
    identifier: str
    license: str = ""
    download_url: str
    thumb: str = ""
    preview_url: str = ""


class AssetSearchResponse(BaseModel):
    clips: list[BrollClip]


class AssembleJobResponse(BaseModel):
    job_id: str


class AssembleStatusResponse(BaseModel):
    status: str
    progress: float
    download_url: str | None = None
    error: str | None = None


class BrollDownloadRequest(BaseModel):
    url: HttpUrl


class BrollDownloadResponse(BaseModel):
    local_path: str
    filename: str
